src/libmongoc/src/mongoc/mongoc-client.c

```diff
@@ -3080,3 +3088,33 @@ mongoc_client_enable_auto_encryption (mongoc_client_t *client,
    }
    return _mongoc_cse_client_enable_auto_encryption (client, opts, error);
 }
+
+bool
+mongoc_client_set_timeout_ms (mongoc_client_t *client,
+			      int64_t timeout,
+			      bson_error_t *error)
+{
+   BSON_ASSERT_PARAM (client);
+
+   if (timeout < 0) {
+      bson_set_error (error,
+		      MONGOC_ERROR_TIMEOUT,
+		      MONGOC_ERROR_TIMEOUT_INVALID,
+		      "timeoutMS must be a non-negative integer");
+	 return false;
+   }
```

Comment:
Done.

Reply:
I agree. Feel free to change whatever. Yeah, this isn't public-facing, so no reason we can't change that type as needed. The only real pain will be updating the expected error messages in `test-mongoc-timeout.c`.

Reply:
> It doesn't really look like mongoc_timeout_t is used elsewhere in the codebase yet, and it's not a public-facing type,

This answers my previous question about the getters returning `int64_t` instead of `mongoc_timeout_t` objects.

If the type is entirely private, would it make sense to use a static struct initializer (e.g. `{0}`) instead of memory allocation in `mongoc_timeout_new()` and just embed the timeout struct in other types? This isn't performance sensitive like BSON and I don't feel strongly about it but wanted to ask in case it was never considered.

Reply:
I didn't consider it. I'd be open to it though.

Reply:
In the interest of getting this code review to the finish line, I'd like to leave `mongoc_timeout_t` as-is for now, but I'd be open to changing this in the future